front/components/assistant_builder/AssistantBuilder.tsx

```diff
@@ -723,25 +718,26 @@ export async function submitAssistantBuilderForm({
     | NonNullable<BodyType["assistant"]["actions"]>[number]
     | null = null;
 
-  switch (builderState.actionMode) {
-    case "GENERIC":
+  const action = getDeprecatedDefaultSingleAction(builderState);
```

Comment:
Why take one if the route accepts an array?

Reply:
I wanted to avoid changing too many things, but indeed seems silly in retrospect

Reply:
Fixed